lib/OpenQA/JobSettings.pm

```diff
@@ -18,7 +18,9 @@ use strict;
 use warnings;
 
 sub generate_settings {
-    my ($params) = @_;
+    my ($params, $destructive, $definitive) = @_;
```

Comment:
As you are already changing the parameter list how about named parameters?:

```perl
    my ($params, %args) = @_;
    $args{destructive} //= 1;
    $args{definitive}  //= 1;
…
```

and calling it like:

```perl
    my $error = OpenQA::JobSettings::generate_settings(\%params, destructive => 0, definitive => 0);
```